### ISSUE TITLE
Misleading error given when using KVS command http.asyncGet without brackets

### ISSUE THREAD

**Issue by DarthGandalf** [kvirc/KVIrc#1472]
**Reported by Omega on 18 Aug 2014 19:40:14 UTC**
'/http.asyncGet' in a server window outputs the following failure:

```
[20:16:16] [KVS] Runtime Error: Module command call failed: the module 'http' doesn't export a command named 'asyncGet'
[20:16:16] [KVS]   in script context "commandline::kvs", line 1, near character 1
```

http.asyncGet is available in the help and is tab-completable as a command.

Tested with r6382

---

Migrated from: https://svn.kvirc.de/kvirc/ticket/1472

**Comment by DarthGandalf** [comment]
**Modified by Omega on 20 Aug 2014 19:02:15**
Summary: KVS command http.asyncGet present but not fully recognised by http module → Misleading error given when using KVS command http.asyncGet without brackets

**Comment by DarthGandalf** [comment]
**Comment by Omega on 20 Aug 2014 19:07:30**
This is caused by primitive parsing code determining whether text is a callback command or a simple command by looking for '(', rather than checking what the command in question is registered as - see 'src/kvirc/kvs/parser/KviKvsParser_command.cpp, KviKvsParser::parseCommand, line 224.

When no '(' is present in the text, it assumes the command is a simple command, and when it finds none registered, results in this error returned. Perhaps the error message should prod the user about this potential problem:

the module 'http' doesn't export a command named 'asyncGet' - possibly missing brackets in a callback command?

**Comment by DarthGandalf** [comment]
**Comment by Omega on 25 Aug 2014 10:42:07**
Fixed in #6384 based on CtrlAltca's diff.

---

Resolution: none → fixed
Status: new → closed
